### PR TITLE
Feature/forms

### LINKED ISSUE #17: As a user, I should be able to change my address
- [x]  address
- [x]  city
- [x] zip
- [x]  submit button
- [x] I shouldn't be able to hit submit if any fields are blank
- [ ] I should be notified what I need to do if a field is blank
- [x] I can only enter 5 digits into zip code input

Acceptance Criteria

- [x] I can click a button and see a form to change location
- [x] I can fill out an address, city, and zip code and submit a location and the form will disappear
- [ ] When I don't fill out all the fields, a message will appear telling me I need to fill out the all fields and I can't submit the form
- [ ] When I submit my location, the map will show me a new marker with the updated location

### LINKED ISSUE #18: As a user, I should be able to create a new login
- [x] first name
- [x] last name
- [x] login button
- [x] new user button
- [x]  I shouldn't be able to hit submit if any fields are blank
- [ ] I should be notified what I need to do if a field is blank

### LINKED ISSUE #21: As a user, I should be be able to log in
- [x] first name
- [x] last name
- [x] login button
- [x] I shouldn't be able to hit submit if any fields are blank
- [ ] I should be notified what I need to do if a field is blank

### PR DESCRIPTION
**Pull request by Shakikka**
## What does this PR do?

- Adds a Form component
- Adds routing and conditional rendering for form contents
- Adds form styling
- Add empty input handling
- Change website name and appropriate auto deploy settings

## How should it be tested?

- on the website type in input fields and click on buttons.

## What do the future iterations hold?

- error messages for empty input fields
- food truck component
- food list component
- better styling

## Linked Issues
- [#21](https://github.com/iscreamtruck/foodtruck-fe/issues/21)
- [#18](https://github.com/iscreamtruck/foodtruck-fe/issues/18)
- [#17](https://github.com/iscreamtruck/foodtruck-fe/issues/17)

### CODE REVIEW

File: public/browserconfig.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<browserconfig>
+    <msapplication>
+        <tile>
+            <square150x150logo src="/mstile-150x150.png"/>
+            <TileColor>#da532c</TileColor>
+        </tile>
+    </msapplication>
+</browserconfig>
```
Comment (elizhahn) [inline public/browserconfig.xml:9]: what is going on here?

File: src/components/App.js [hunk 2]
```diff
@@ -35,12 +36,29 @@ class App extends Component {
...
+          <Route exact path='/login'>
+            <Form />
+          </Route>
+          <Route exact path="/newuser">
+            <Form/>
+          </Route>
+          <Route exact path="/newlocation">
+            <Form/>
+          </Route>
+          <Route exact path="/trucklist">
```
Comment (elizhahn) [inline src/components/App.js:52]: Do we need exact path for all of these?

File: src/components/Form/Form.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    const handleSubmit = e => {
+        if (location === '/newuser' && first && last && address && city && zip.length === 5) {
+            clearInputs()
+        } else if (location === '/login' && userName) {
+            clearInputs()
+        } else if (location === '/newlocation' && address && city && zip.length === 5) {
+            clearInputs()
+        } else {
+            e.preventDefault()
+        }
```
Comment (elizhahn) [inline src/components/Form/Form.js:22]: won't we need e.preventDefault() for all of these conditions? should we stick it at the top?

File: public/browserconfig.xml (hunk 1 above)
Comment (Shakikka) [reply public/browserconfig.xml:9]: This was one of the imported files from creating our new favicon. I'm not entirely sure what it is doing exactly.

Comment (elizhahn) [reply public/browserconfig.xml:9]: ok that makes sense, thanks

File: src/components/Form/Form.js (hunk 3 above)
Comment (Shakikka) [reply src/components/Form/Form.js:22]: Having `e.preventDefault()` at the top was preventing it from routing correctly, so I think the `Link` replaced the default action of the button to just change the route. We may have to change it in the future, but it is working as is currently.

File: src/components/App.js (hunk 2 above)
Comment (Shakikka) [reply src/components/App.js:52]: I'm not sure. Do we not need exact path for all of these?

Comment (elizhahn) [reply src/components/App.js:52]: haha, well I think its fine either way, but something to look into on best practices maybe. Let's keep it the way it is

File: src/components/Form/Form.js (hunk 3 above)
Comment (elizhahn) [reply src/components/Form/Form.js:22]: oh ok, that's right, I remember that routing prevented refreshing and makes for a smoother UX. That makes sense thanks

**Pull request merged**